src/utils.ts

```diff
@@ -48,17 +48,25 @@ export const hasOauthClientSettings = (local = false): boolean =>
  * Should be used instead of `DOTFILE.RC?().read()`
  * @returns {Promise<ClaspToken>} A promise to get the rc file as object.
  */
-export function getOAuthSettings(): Promise<ClaspToken> {
-  return DOTFILE.RC_LOCAL()
-    .read()
-    .then((rc: ClaspToken) => rc)
-    .catch((err: any) => {
-      return DOTFILE.RC.read()
-        .then((rc: ClaspToken) => rc)
-        .catch((err: any) => {
-          logError(err, ERROR.NO_CREDENTIALS);
-        });
-    });
+export function getOAuthSettings(localCredsOnly: boolean): Promise<ClaspToken> {
+  if (localCredsOnly) {
+    return DOTFILE.RC_LOCAL().read()
+      .then((rc: ClaspToken) => rc)
+      .catch((err: any) => {
+        logError(err, ERROR.NO_CREDENTIALS(localCredsOnly));
+      });
+  } else {
+    return DOTFILE.RC_LOCAL()
```

Comment:
We should be exclusive.
- local creds are **only** used for `clasp run`
- global creds are used for everything else

I previously thought that local creds should be used for everything once you use them, but talking to some senior eng's at Google convinced me that we should do the above (local only for run).

Reply:
What about the case where you are managing 2 different `clasp` projects (with different creds)?

```
~/project_1/.clasprc.json
~/project_2/.clasprc.json
```

would work with the recursive strategy.

However, with just

```
~/.clasprc.json
```

Users would have to log back in and out each time.

Reply:
True. That's been an issue. #42
I updated the comment: ~global~ recursive-parent

For the multi-login issue, I think that should be separate from cloud projects.

Something like:

```
clasp login --local
```

This would use global `clasp` credentials but store a token locally.

```
clasp login # global
clasp login --creds creds.json # your own creds
clasp login --local # local
clasp login --creds creds.json --local # own creds, local
```

Reply:
Ah, I had not seen your comment there yet. I like Workaround 2 as well as the solution proposed here.

Note: does recursive-parent creds include cwd, or is it just parent and above?

Reply:
local is inclusive:

```
/clasptest/ $ clasp login --local # Log in to Gmail 1
/clasptest/ $ clasp create
/clasptest/ $ echo "function hello() { return 'hi' };" > hello.js
/clasptest/ $ clasp push
```

Reply:
@grant do you want me to include `--local` in this PR? Or, should I just change the `else {` to check the global creds only?

Reply:
If we're going to modify login, I'd like to see a change moving towards a `--local` option to solve the variety of issues around multi-login.

Reply:
Starting with a new PR would be OK.